sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+//
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {
+  WebResource,
+  ServiceClient,
+  ServiceClientOptions,
+  HttpOperationResponse,
+  SasServiceClientCredentials,
+  AtomXmlOperationSpec,
+  signingPolicy,
+  userAgentPolicy,
+  logPolicy,
+  proxyPolicy,
+  atomSerializationPolicy,
+  RequestPolicyFactory,
+  ResourceSerializer,
+  URLBuilder,
+  ProxySettings
+} from "@azure/core-http";
+
+import { httpAtomXml } from "./log";
+
+import { QueueResourceSerializer } from "./serializers/queueResourceSerializer";
+import { TopicResourceSerializer } from "./serializers/topicResourceSerializer";
+import { SubscriptionResourceSerializer } from "./serializers/subscriptionResourceSerializer";
+import { RuleResourceSerializer } from "./serializers/ruleResourceSerializer";
+
+/**
+ * All operations return a `Promise<HttpOperationResponse>` object of which the `parsedBody`
+ * is populated with the response in JSON format with following structure:
+ *  {
+ *   response: <actual response>,
+ *   result: <parsed result, if any>,
+ *   error: <error information, if any>
+ *  }
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  endpoint: any;
```

Comment:
Is `endpoint` supposed to be public? And can it be a type other than string? (it looks like you're setting it to a string in the constructor).

Reply:
This should be private and typed to string